UpgradeClusterClientYamlTestSuiteIT tests fail because of deprecation warnings

Came across this in a PR build: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request-2/7278/console

Example output:
```
3:04:47 ERROR   0.08s | UpgradeClusterClientYamlTestSuiteIT.test {p0=old_cluster/40_ml_datafeed_crud/Put job and datafeed in old cluster} <<< FAILURES!
13:04:47    > Throwable #1: org.elasticsearch.client.WarningFailureException: method [GET], host [http://[::1]:46781], URI [/_template/.ml-notifications], status line [HTTP/1.1 200 OK]
13:04:47    > Warnings: [[types removal] The response format of get index template requests will change in the next major version. Please start using the `include_type_name` parameter set to `false` in the request to move to the new, typeless response format that will be the default in 7.0.]
13:04:47    > {".ml-notifications":{"order":0,"version":6070099,"index_patterns":[".ml-notifications"],"settings":{"index":{"number_of_shards":"1","auto_expand_replicas":"0-1","unassigned":{"node_left":{"delayed_timeout":"1m"}}}},"mappings":{"audit_message":{"properties":{"job_id":{"type":"keyword"},"level":{"type":"keyword"},"message":{"type":"text","fields":{"raw":{"type":"keyword"}}},"timestamp":{"type":"date"},"node_name":{"type":"keyword"}}}},"aliases":{}}}
13:04:47    > 	at __randomizedtesting.SeedInfo.seed([CA5B3D2C2AEFD704:420F02F68413BAFC]:0)
13:04:47    > 	at org.elasticsearch.client.RestClient$SyncResponseListener.get(RestClient.java:690)
13:04:47    > 	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:218)
13:04:47    > 	at org.elasticsearch.xpack.test.rest.XPackRestTestHelper.lambda$waitForTemplates$1(XPackRestTestHelper.java:79)
13:04:47    > 	at org.elasticsearch.test.ESTestCase.awaitBusy(ESTestCase.java:879)
13:04:47    > 	at org.elasticsearch.test.ESTestCase.awaitBusy(ESTestCase.java:868)
13:04:47    > 	at org.elasticsearch.xpack.test.rest.XPackRestTestHelper.waitForTemplates(XPackRestTestHelper.java:76)
13:04:47    > 	at org.elasticsearch.upgrades.UpgradeClusterClientYamlTestSuiteIT.waitForTemplates(UpgradeClusterClientYamlTestSuiteIT.java:31)
13:04:47    > 	at java.lang.Thread.run(Thread.java:748)
13:04:47    > Caused by: org.elasticsearch.client.WarningFailureException: method [GET], host [http://[::1]:46781], URI [/_template/.ml-notifications], status line [HTTP/1.1 200 OK]
13:04:47    > Warnings: [[types removal] The response format of get index template requests will change in the next major version. Please start using the `include_type_name` parameter set to `false` in the request to move to the new, typeless response format that will be the default in 7.0.]
13:04:47    > {".ml-notifications":{"order":0,"version":6070099,"index_patterns":[".ml-notifications"],"settings":{"index":{"number_of_shards":"1","auto_expand_replicas":"0-1","unassigned":{"node_left":{"delayed_timeout":"1m"}}}},"mappings":{"audit_message":{"properties":{"job_id":{"type":"keyword"},"level":{"type":"keyword"},"message":{"type":"text","fields":{"raw":{"type":"keyword"}}},"timestamp":{"type":"date"},"node_name":{"type":"keyword"}}}},"aliases":{}}}
13:04:47    > 	at org.elasticsearch.client.RestClient$1.completed(RestClient.java:304)
13:04:47    > 	at org.elasticsearch.client.RestClient$1.completed(RestClient.java:294)
13:04:47    > 	at org.apache.http.concurrent.BasicFuture.completed(BasicFuture.java:122)
13:04:47    > 	at org.apache.http.impl.nio.client.DefaultClientExchangeHandlerImpl.responseCompleted(DefaultClientExchangeHandlerImpl.java:181)
13:04:47    > 	at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.processResponse(HttpAsyncRequestExecutor.java:448)
13:04:47    > 	at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.inputReady(HttpAsyncRequestExecutor.java:338)
13:04:47    > 	at org.apache.http.impl.nio.DefaultNHttpClientConnection.consumeInput(DefaultNHttpClientConnection.java:265)
13:04:47    > 	at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:81)
13:04:47    > 	at org.apache.http.impl.nio.client.InternalIODispatch.onInputReady(InternalIODispatch.java:39)
13:04:47    > 	at org.apache.http.impl.nio.reactor.AbstractIODispatch.inputReady(AbstractIODispatch.java:114)
13:04:47    > 	at org.apache.http.impl.nio.reactor.BaseIOReactor.readable(BaseIOReactor.java:162)
13:04:47    > 	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvent(AbstractIOReactor.java:337)
13:04:47    > 	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvents(AbstractIOReactor.java:315)
13:04:47    > 	at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:276)
13:04:47    > 	at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)
13:04:47    > 	at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:591)
13:04:47    > 	... 1 more
```

This looks similar to https://github.com/elastic/elasticsearch/issues/38395 and looks like it can be resolved by dealing with the deprecation warnings in the [XPackRestTestHelper](https://github.com/elastic/elasticsearch/blob/5384162a42b066ee4bcadba6ab6d699f6966747f/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/test/rest/XPackRestTestHelper.java#L79)

--- a/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/test/rest/XPackRestTestHelper.java
+++ b/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/test/rest/XPackRestTestHelper.java
@@ -14,6 +14,7 @@ import org.elasticsearch.client.RestClient;
 import org.elasticsearch.common.xcontent.XContentHelper;
 import org.elasticsearch.common.xcontent.json.JsonXContent;
 import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.test.rest.ESRestTestCase;
 import org.elasticsearch.xpack.core.ml.MlMetaIndex;
 import org.elasticsearch.xpack.core.ml.job.persistence.AnomalyDetectorsIndex;
 import org.elasticsearch.xpack.core.ml.job.persistence.AnomalyDetectorsIndexFields;
@@ -76,7 +77,9 @@ public final class XPackRestTestHelper {
             ESTestCase.awaitBusy(() -> {
                 Map<?, ?> response;
                 try {
-                    String string = EntityUtils.toString(client.performRequest(new Request("GET", "/_template/" + template)).getEntity());
+                    final Request getRequest = new Request("GET", "_template/" + template);
+                    getRequest.setOptions(ESRestTestCase.allowTypeRemovalWarnings());
+                    String string = EntityUtils.toString(client.performRequest(getRequest).getEntity());
                     response = XContentHelper.convertToMap(JsonXContent.jsonXContent, string, false);
                 } catch (ResponseException e) {
                     if (e.getResponse().getStatusLine().getStatusCode() == 404) {

